wgsl/index.bs

```diff
@@ -689,84 +682,240 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
-* OpenGL `std140` layout of `mat2x2` has extra padding between column vectors that is not present in a `mat2x2` type in [SHORTNAME].
-    * The OpenGL `std140` layout of the `mat2x2` type has the second column vector starting 16 bytes after the first column vector.
-        But in [SHORTNAME] the second column vector of `mat2x2<f32>` starts 8 bytes after the first column vector.
-* For any type, OpenGL `std430` layout is the same as using the tightest offset and stride assignments in [SHORTNAME] [=storage buffer layout=].
-* OpenGL supports row-major matrices, but [SHORTNAME] does not.
-
-Compared to [[Vulkan1.2ext|Vulkan &sect; 15.6.4 Offset and Stride Assignment]]:
-* Vulkan *standard buffer layout* maps to [SHORTNAME] [=standard buffer layout rules=] with the following qualifications:
-    * Vulkan allows a vector to be aligned to the size of its scalar component, but [SHORTNAME] requires a more constrained alignment.
-    * The Vulkan `scalarBlockLayout` and `uniformBufferStandardLayout` features do not apply to [SHORTNAME].
-    * The Vulkan concept of *scalar alignment* does not correspond to a concept in [SHORTNAME];
-* The Vulkan *base alignment* for a type |S| corresponds to the [SHORTNAME] alignment requirement for |S| in the [=storage classes/storage=] storage class: |Align|(|S|,`storage`).
-* The Vulkan *extended alignment* for a type |S| corresponds to the [SHORTNAME] alignment requirement for |S| in the [=storage classes/uniform=] storage class: |Align|(|S|,`uniform`).
-* The Vulkan concept of *improperly straddle* is not permitted in [SHORTNAME], because [SHORTNAME] requires vectors to be aligned to their whole size.
-* Vulkan supports non-32-bit scalar types and vector types with non-32-bit components, but [SHORTNAME] does not.
-* Vulkan supports row-major matrices, but [SHORTNAME] does not.
-* Vulkan allows offsets to be non-monotonic, but [SHORTNAME] does not.
-
-#### Internal Layout of Values #### {#internal-value-layout}
+Structure members must not overlap. Structure members decorated with the
+[=size=] attribute must have a size at least as large as the size of the
+member's type.
+
+The alignment of a structure is equal to the largest alignment of all of its
+members:
+<p algorithm="structure alignment">
+  [=AlignOf=](|S|) = max([=AlignOf=](|S|, M<sub>0</sub>), ... , [=AlignOf=](|S|, M<sub>n</sub>))
+</p>
+
+The size of a structure is equal to the offset plus the size of its last member,
+rounded to the next multiple of the structure's alignment:
+<p algorithm="structure size">
+  [=SizeOf=](|S|) = [=roundUp=]([=AlignOf=](|S|), [=OffsetOf=](|S|, |L|) + [=SizeOf=](|S|, |L|))<br>
+  Where |L| is the last member of the structure
+</p>
+
+<div class='example wgsl' heading='Layout of structures using implicit member strides and alignments'>
+  <xmp highlight='rust'>
+    /*             align(8)  size(24)  */ [[block]] struct A {
```

Comment:
WGSL doesn't have `/*` comments, only `// comments. These should probably move to the line above what they're annotating.

Reply:
Done